app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt

```diff
@@ -84,11 +79,14 @@ open class SearchAllPeopleViewModel(
         }
     }
 
-    override suspend fun getInitialContacts(): SearchResult =
-        when (val result = getAllKnownUsers()) {
-            is GetAllContactsResult.Failure -> SearchResult.Failure(R.string.label_general_error)
-            is GetAllContactsResult.Success -> SearchResult.Success(result.allContacts.map(contactMapper::fromOtherUser))
+    override suspend fun getInitialContacts(): Flow<SearchResult> = getAllKnownUsers()
+        .map { result ->
+            when (result) {
+                is GetAllContactsResult.Failure -> SearchResult.Failure(R.string.label_general_error)
+                is GetAllContactsResult.Success -> SearchResult.Success(result.allContacts.map(contactMapper::fromOtherUser))
+            }
         }
+        .flowOn(dispatcher.io())
```

Comment:
I do not think there is need for a change to a function, you can keep 

protected val initialContactResultFlow as the rest as a variable, and keep the rest you did with a map { }

Reply:
Same here